tests/ignite/metrics/test_entropy.py

```diff
@@ -0,0 +1,206 @@
+        (torch.randn((100, 5, 224, 224)), torch.randint(0, 5, size=(100, 224, 224)), 16)
+    ][request.param]
+
+
+@pytest.mark.parametrize("n_times", range(5))
+def test_compute(n_times, test_case):
+    ent = Entropy()
+
+    y_pred, y, batch_size = test_case
+
+    ent.reset()
+    if batch_size > 1:
+        n_iters = y.shape[0] // batch_size + 1
+        for i in range(n_iters):
+            idx = i * batch_size
+            ent.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
+    else:
+        ent.update((y_pred, y))
+
+    np_res = np_entropy(y_pred.numpy())
+
+    assert isinstance(ent.compute(), float)
+    assert pytest.approx(ent.compute()) == np_res
+
+
+def _test_distrib_integration(device, tol=1e-6):
+    from ignite.engine import Engine
+
+    rank = idist.get_rank()
+    torch.manual_seed(12 + rank)
+
+    def _test(metric_device):
+        n_iters = 100
+        batch_size = 10
+        n_cls = 50
+
+        y_true = torch.randint(0, n_cls, size=[n_iters * batch_size], dtype=torch.long).to(device)
+        y_preds = torch.normal(2.0, 3.0, size=(n_iters * batch_size, n_cls), dtype=torch.float).to(device)
+
+        def update(engine, i):
+            return (
+                y_preds[i * batch_size : (i + 1) * batch_size],
+                y_true[i * batch_size : (i + 1) * batch_size],
+            )
+
+        engine = Engine(update)
+
+        m = Entropy(device=metric_device)
+        m.attach(engine, "entropy")
+
+        data = list(range(n_iters))
+        engine.run(data=data, max_epochs=1)
+
+        y_preds = idist.all_gather(y_preds)
+        y_true = idist.all_gather(y_true)
+
+        assert "entropy" in engine.state.metrics
+        res = engine.state.metrics["entropy"]
+
+        true_res = np_entropy(y_preds.numpy())
+
+        assert pytest.approx(res, rel=tol) == true_res
+
+    _test("cpu")
+    if device.type != "xla":
+        _test(idist.device())
+
+
+def _test_distrib_accumulator_device(device):
+    metric_devices = [torch.device("cpu")]
+    if device.type != "xla":
+        metric_devices.append(idist.device())
+    for metric_device in metric_devices:
+        device = torch.device(device)
+        ent = Entropy(device=metric_device)
+
+        for dev in [ent._device, ent._sum_of_entropies.device]:
+            assert dev == metric_device, f"{type(dev)}:{dev} vs {type(metric_device)}:{metric_device}"
+
+        y_pred = torch.tensor([[2.0], [-2.0]])
+        y = torch.zeros(2)
+        ent.update((y_pred, y))
+
+        for dev in [ent._device, ent._sum_of_entropies.device]:
+            assert dev == metric_device, f"{type(dev)}:{dev} vs {type(metric_device)}:{metric_device}"
+
+
+def test_accumulator_detached():
+    ent = Entropy()
+
+    y_pred = torch.tensor([[2.0, 3.0], [-2.0, -1.0]], requires_grad=True)
+    y = torch.zeros(2)
+    ent.update((y_pred, y))
+
+    assert not ent._sum_of_entropies.requires_grad
+
+
+@pytest.mark.distributed
```

Comment:
For distributed config tests, could you please rewrite them using new testing formalism that we are trying to adopt. Here is an example of the code to inspire of:
https://github.com/pytorch/ignite/blob/5fe74432ed5813d803b5c40641cd47d0cb9964df/tests/ignite/metrics/test_recall.py#L422-L426

Here is a PR showing how to pass from old code to the new one:
https://github.com/pytorch/ignite/pull/3208/files#diff-c56c264ef288f88e5738e9ad22de66dffd4c58d2e656eb62e8dbaa678672317d

Thanks!